apache2/msc_multipart.c

```diff
@@ -130,7 +134,41 @@ static int multipart_parse_content_disposition(modsec_rec *msr, char *c_d_value)
          */
 
         char quote = '\0';
-        if ((*p == '"') || (*p == '\'')) {
+        if (name == "filename*") {
+            /* filename*=charset'[optional-language]'filename */
+            /* Read beyond the charset and the optional language*/
+            const char* start_of_charset = p;
+            while ((*p != '\0') && (isalnum(*p) || (strchr(mime_charset_special, *p)))) {
+                p++;
+            }
+            if ((*p != '\'') || (p == start_of_charset)) {
+                return -16; // Must be at least one legit char before ' for start of language
+            }
+            p++;
+            while ((*p != '\0') && (*p != '\'')) {
+                p++;
+            }
+            if (*p != '\'') {
+                return -17; // Single quote for end-of-language not found
+            }
+            p++;
+
+            /* Now read what should be the actual filename */
+            const char* start_of_filename = p;
```

Comment:
Looking at the parsing of `filename` below, I expect the corresponding logic here to include the same use of flags and logging, e.g., `msr_log(msr, 4, "Multipart: Warning: Duplicate Content-Disposition name: %s",`.

Reply:
Do you think about [this](https://github.com/owasp-modsecurity/ModSecurity/blob/524a7996b010ebae64f51b1c5d36dbb150c5ef08/apache2/msc_multipart.c#L266) line? (Line 266. in current implementation):

```C
                msr_log(msr, 4,
                    "Multipart: Warning: Duplicate Content-Disposition filename*: %s.",  log_escape_nq(msr->mp, value));
                return -20;
```

Reply:
Never mind. I was confused here as well about the two if/else.